django/db/backends/base/schema.py

```diff
@@ -129,17 +140,25 @@ def execute(self, sql, params=()):
             )
         # Account for non-string statement objects.
         sql = str(sql)
+        prepared_params = None
+        if params is not None:
+            prepared_params = [self.prepare_param(p) for p in params]
+            params_sql = tuple(filter(None, (p[0] for p in prepared_params)))
+            if params_sql:
+                sql %= params_sql
+            prepared_params = tuple(q for p in prepared_params for q in p[1])
+
```

Comment:
Feels like this should not be handled by `execute` but by its callers that need to pass unprepared params.

Was this to avoid updating `create_model`, `add_field`, and `alter_field` paths?

Reply:
Should it, I don't know? It would cause a lot of duplication? You are Mr. Database, so I'd follow your advice. Any concrete suggestions where to put this?